packages/base-driver/lib/basedriver/commands/execute.js

```diff
@@ -0,0 +1,61 @@
+// @ts-check
+import _ from 'lodash';
+import {makeArgs, checkParams} from '../../protocol';
+
+/**
+ * @param {SessionBase} Base
+ * @returns {ExecuteBase}
+ */
+export function ExecuteMixin(Base) {
+  /**
+   * @implements {IExecuteCommands}
+   */
+  class ExecuteCommands extends Base {
+    /**
+     * @this {Driver}
+     * @param {string} script
+     * @param {[Record<string, any>]|[]} protoArgs
+     */
+    async executeMethod(script, protoArgs) {
+      // the w3c protocol will give us an array of arguments to apply to a javascript function.
+      // that's not what we're doing. we're going to look for a JS object as the first arg, so we
+      // can perform validation on it. we'll ignore everything else.
+      let args = protoArgs[0];
+      if (!args) {
+        args = {};
+      }
+      if (!_.isPlainObject(args)) {
+        throw new Error(`Did not receive an appropriate executeMethod parameters object`);
+      }
+
+      const Driver = /** @type {DriverClass} */ (this.constructor);
+      const availableScripts = _.keys(Driver.executeMethodMap);
+      const commandMetadata = Driver.executeMethodMap[script];
+      if (!commandMetadata) {
+        throw new Error(
```

Comment:
maybe we can throw a more specific W3C error?

Reply:
yep, found some more specific protocol errors.